File: Core/Domain/Entities/Department.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Department:BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }

        public int CompanyId { get; set; }
        public Company Company { get; set; }
        public List<Employee> Employees { get; set; }
    }
}

File: client/src/app/shared/models/department.ts
export interface IDepartment {
  id: number;
  name: string;
  description: string;
  companyId: number;
}
export class Department implements IDepartment {
  id: number;
  name: string;
  description: string;
  companyId: number;
  constructor(
    id: number,
    name: string,
    description: string,
    companyId: number
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.companyId = companyId;
  }
}
